ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/builders/JarManifestErrorReporter.java

```diff
@@ -61,6 +62,15 @@ private String getHeaderName(String line) {
 		return null;
 	}
 
+	private boolean isInvalidHeader(String line) {
+		int colonInd = line.indexOf(':');
+		if (colonInd > 0) {
+			String headerPart = line.substring(0, colonInd);
+			return headerPart != null && headerPart.matches(".*[^a-zA-Z0-9-_].*"); //$NON-NLS-1$
```

Comment:
String.substring doesn't return null - either returns a valid string or throws OOB exception. given you are checking the bounds already, I think the null check is redundant?